File: runme/mcgen/local/datan/exasrc/e1sd.c
#include <stdio.h>
#include "datsrc.h"

main(void)
{
    static integer k, n, iswit, iswit1, kk, nn;
    static double f, p, alambd;
/* identify program to user */
    printf("%s\n", "Program E1SD demonstrates use of");
    printf("%s\n", "SDBINM,SCBINM,SDHYPG,SCHYPG,");
    printf("%s\n", "SDPOIS,SCPOIS,SQPOIS");
    printf("%s\n", "and provides interactively numerical values");
    printf("%s\n\n", "of statistical functions of a discrete variable.");
    printf("%s\n", "Choose:");
    printf("%s\n", "1 - binomial distribution");
    printf("%s\n", "2 - hypergeometric distribution");
    printf("%s\n", "3 - Poisson distribution");
    printf("%s", "> ");
    scanf("%li", &iswit);
    if (iswit == 1) {
/* binomial */
        printf("%s\n", "Choose:");
        printf("%s\n", "1 - probability");
        printf("%s\n", "2 - distribution function (cumulative prob.)");
        printf("%s", "> ");
        scanf("%li", &iswit1);
        if (iswit1 == 1) {
            printf("%s\n", "Enter p (0.<p<1.):");
            printf("%s", "> ");
            scanf("%lf", &p);
            printf("%s\n", "Enter n (>=0):");
            printf("%s", "> ");
            scanf("%li", &n);
            printf("%s\n", "Enter k (0<=k<=n):");
            printf("%s", "> ");
            scanf("%li", &k);
	    f = sdbinm_(&k, &n, &p);
            printf("%s%14.10lf\n", "W(n,k) = ", f);
	} else if (iswit1 == 2) {
            printf("%s\n", "Enter p (0.<p<1.):");
            printf("%s", "> ");
            scanf("%lf", &p);
            printf("%s\n", "Enter n (>=0):");
            printf("%s", "> ");
            scanf("%li", &n);
            printf("%s\n", "Enter K (0<=k<=n):");
            printf("%s", "> ");
            scanf("%li", &k);
	    f = scbinm_(&k, &n, &p);
            printf("%s%14.10lf\n", "P(k<K) = ", f);
	}
    } else if (iswit == 2) {
        printf("%s\n", "Choose:");
        printf("%s\n", "1 - probability");
        printf("%s\n", "2 - distribution function (cumulative prob.)");
        printf("%s", "> ");
        scanf("%li", &iswit1);
	if (iswit1 == 1) {
            printf("%s\n", "Enter K (>=0):");
            printf("%s", "> ");
            scanf("%li", &kk);
            printf("%s\n", "Enter N (>=K):");
            printf("%s", "> ");
            scanf("%li", &nn);
            printf("%s\n", "Enter k (0<=k<=K):");
            printf("%s", "> ");
            scanf("%li", &k);
            printf("%s\n", "Enter n (k<=n<N):");
            printf("%s", "> ");
            scanf("%li", &n);
	    f = sdhypg_(&k, &n, &kk, &nn);
            printf("%s%14.10lf\n", "W(k) = ", f);
	} else if (iswit1 == 2) {
            printf("%s\n", "Enter K (>=0):");
            printf("%s", "> ");
            scanf("%li", &kk);
            printf("%s\n", "Enter N (>=K):");
            printf("%s", "> ");
            scanf("%li", &nn);
            printf("%s\n", "Enter kprime (<=K):");
            printf("%s", "> ");
            scanf("%li", &k);
            printf("%s\n", "Enter n (<=N):");
            printf("%s", "> ");
            scanf("%li", &n);
	    f = schypg_(&k, &n, &kk, &nn);
            printf("%s%14.10lf\n", "P(k<kprime) = ", f);
	}
    } else if (iswit == 3) {
        printf("%s\n", "Choose:");
        printf("%s\n", "1 - probability");
        printf("%s\n", "2 - distribution function (cumulative prob.)");
        printf("%s\n", "3 - quantile");
        printf("%s\n", ">");
        scanf("%li", &iswit1);
	if (iswit1 == 1) {
            printf("%s\n", "Enter k (>=0):");
            printf("%s", "> ");
            scanf("%li", &k);
            printf("%s\n", "Enter lambda (>0.):");
            printf("%s", "> ");
            scanf("%lf", &alambd);
	    f = sdpois_(&k, &alambd);
            printf("%s%14.10lf\n", "f(k;lambda) = ", f);
	} else if (iswit1 == 2) {
            printf("%s\n", "Enter K (>=0):");
            printf("%s", "> ");
            scanf("%li", &kk);
            printf("%s\n", "Enter lambda (>0.):");
            printf("%s", "> ");
            scanf("%lf", &alambd);
	    f = scpois_(&kk, &alambd);
            printf("%s%14.10lf\n", "P(k<K) = ", f);
	} else if (iswit1 == 3) {
            printf("%s\n", "Enter K (>=0):");
            printf("%s", "> ");
            scanf("%li", &kk);
            printf("%s\n", "Enter P (0.<P<1.):");
            printf("%s", "> ");
            scanf("%lf", &p);
	    f = sqpois_(&kk, &p);
            printf("%s%14.10lf\n", "lambda_P (K) = ", f);
	}
    }
    return 0;
} /* main */

File: runme/mcgen/local/datan/exasrc/e1sd.f
      PROGRAM E1SD
      IMPLICIT DOUBLE PRECISION(A-H,O-Z)
C identify program to user
      WRITE(*,*)' Program E1SD demonstrates use of'
      WRITE(*,*)' SDBINM,SCBINM,SDHYPG,SCHYPG,'
      WRITE(*,*)' SDPOIS,SCPOIS,SQPOIS'
      WRITE(*,*)' and provides interactively numerical values'
      WRITE(*,*)' of statistical functions of a discrete variable'
      WRITE(*,*)' '
      WRITE(*,*)' Choose'
      WRITE(*,*)' 1 - binomial distribution'
      WRITE(*,*)' 2 - hypergeometric distribution'
      WRITE(*,*)' 3 - Poisson distribution'
      WRITE(*,*)' >'
      READ(*,*)ISWIT
      IF(ISWIT.EQ.1) THEN
C binomial
        WRITE(*,*)' Choose'
        WRITE(*,*)' 1 - probability'
        WRITE(*,*)' 2 - distribution function (cumulative prob.)'
        WRITE(*,*)' >'
        READ(*,*)ISWIT1
        IF(ISWIT1.EQ.1) THEN
          WRITE(*,*)' Enter p (0.<p<1.)'
          WRITE(*,*)' >'
          READ(*,*)P
          WRITE(*,*)' Enter n (>=0)'
          WRITE(*,*)' >'
          READ(*,*)N
          WRITE(*,*)' Enter k (0<=k<=n)'
          WRITE(*,*)' >'
          READ(*,*)K
          F=SDBINM(K,N,P)
          WRITE(*,*)' W(n,k) = ',F
        ELSE IF(ISWIT1.EQ.2) THEN
          WRITE(*,*)' Enter p (0.<p<1.)'
          WRITE(*,*)' >'
          READ(*,*)P
          WRITE(*,*)' Enter n (>=0)'
          WRITE(*,*)' >'
          READ(*,*)N
          WRITE(*,*)' Enter K (0<=k<=n)'
          WRITE(*,*)' >'
          READ(*,*)K
          F=SCBINM(K,N,P)
          WRITE(*,*)' P(k<K) = ',F
        END IF
      ELSE IF(ISWIT.EQ.2) THEN
        WRITE(*,*)' Choose'
        WRITE(*,*)' 1 - probability'
        WRITE(*,*)' 2 - distribution function (cumulative prob.)'
        WRITE(*,*)' >'
        READ(*,*)ISWIT1
        IF(ISWIT1.EQ.1) THEN
          WRITE(*,*)' Enter K (>=0)'
          WRITE(*,*)' >'
          READ(*,*)KK
          WRITE(*,*)' Enter N (>=K)'
          WRITE(*,*)' >'
          READ(*,*)NN
          WRITE(*,*)' Enter k (0<=k<=K)'
          WRITE(*,*)' >'
          READ(*,*)K
          WRITE(*,*)' Enter n (k<=n<N)'
          WRITE(*,*)' >'
          READ(*,*)N
          F=SDHYPG(K,N,KK,NN)
          WRITE(*,*)' W(k) = ',F
        ELSE IF(ISWIT1.EQ.2) THEN
          WRITE(*,*)' Enter K (>=0)'
          WRITE(*,*)' >'
          READ(*,*)KK
          WRITE(*,*)' Enter N (>=K)'
          WRITE(*,*)' >'
          READ(*,*)NN
          WRITE(*,*)' Enter kprime (<=K)'
          WRITE(*,*)' >'
          READ(*,*)K
          WRITE(*,*)' Enter n (<=N)'
          WRITE(*,*)' >'
          READ(*,*)N
          F=SCHYPG(K,N,KK,NN)
          WRITE(*,*)' P(k<kprime) = ',F
        END IF
      ELSE IF(ISWIT.EQ.3) THEN
        WRITE(*,*)' Choose'
        WRITE(*,*)' 1 - probability'
        WRITE(*,*)' 2 - distribution function (cumulative prob.)'
        WRITE(*,*)' 3 - quantile'
        WRITE(*,*)' >'
        READ(*,*)ISWIT1
        IF(ISWIT1.EQ.1) THEN
          WRITE(*,*)' Enter k (>=0)'
          WRITE(*,*)' >'
          READ(*,*)K
          WRITE(*,*)' Enter lambda (>0.)'
          WRITE(*,*)' >'
          READ(*,*)ALAMBD
          F=SDPOIS(K,ALAMBD)
          WRITE(*,*)' f(k;lambda) = ',F
        ELSE IF(ISWIT1.EQ.2) THEN
          WRITE(*,*)' Enter K (>=0)'
          WRITE(*,*)' >'
          READ(*,*)KK
          WRITE(*,*)' Enter lambda (>0.)'
          WRITE(*,*)' >'
          READ(*,*)ALAMBD
          F=SCPOIS(KK,ALAMBD)
          WRITE(*,*)' P(k<K) = ',F
        ELSE IF(ISWIT1.EQ.3) THEN
          WRITE(*,*)' Enter K (>=0)'
          WRITE(*,*)' >'
          READ(*,*)KK
          WRITE(*,*)' Enter P (0.<P<1.)'
          WRITE(*,*)' >'
          READ(*,*)P
          F=SQPOIS(KK,P)
          WRITE(*,*)' lambda_P (K) = ',F
        END IF
      END IF
      END
